eZ/Publish/Core/Repository/PermissionService.php

```diff
@@ -0,0 +1,279 @@
+                $limitations = $policy->getLimitations();
+
+                /*
+                 * Return true if policy gives full access (aka no limitations)
+                 */
+                if ($limitations === '*') {
+                    return true;
+                }
+
+                /*
+                 * Loop over limitations, all must return ACCESS_GRANTED for policy to pass.
+                 * If limitations was empty array this means same as '*'
+                 */
+                $limitationsPass = true;
+                foreach ($limitations as $limitation) {
+                    $type = $this->limitationService->getLimitationType($limitation->getIdentifier());
+                    $accessVote = $type->evaluate($limitation, $currentUserRef, $object, $targets);
+                    /*
+                     * For policy limitation atm only support ACCESS_GRANTED
+                     *
+                     * Reasoning: Right now, use of a policy limitation not valid for a policy is per definition a
+                     * BadState. To reach this you would have to configure the "policyMap" wrongly, like using
+                     * Node (Location) limitation on state/assign. So in this case Role Limitations will return
+                     * ACCESS_ABSTAIN (== no access here), and other limitations will throw InvalidArgument above,
+                     * both cases forcing dev to investigate to find miss configuration. This might be relaxed in
+                     * the future if valid use cases for ACCESS_ABSTAIN on policy limitations becomes known.
+                     */
+                    if ($accessVote !== LimitationType::ACCESS_GRANTED) {
+                        $limitationsPass = false;
+                        break;// Break to next policy, all limitations must pass
+                    }
+                }
+                if ($limitationsPass) {
+                    return true;
+                }
+            }
+        }
+
+        return false;// None of the limitation sets wanted to let you in, sorry!
+    }
+
+    /**
+     * Allows API execution to be performed with full access sand-boxed.
+     *
+     * The closure sandbox will do a catch all on exceptions and rethrow after
+     * re-setting the sudo flag.
+     *
+     * Example use:
+     *     $location = $repository->sudo(
+     *         function ( Repository $repo ) use ( $locationId )
+     *         {
+     *             return $repo->getLocationService()->loadLocation( $locationId )
+     *         }
+     *     );
+     *
+     *
+     * @param \Closure $callback
+     * @param \eZ\Publish\API\Repository\Repository $outerRepository
+     *
+     * @throws \RuntimeException Thrown on recursive sudo() use.
+     * @throws \Exception Re throws exceptions thrown inside $callback
+     *
+     * @return mixed
+     */
+    public function sudo(\Closure $callback, RepositoryInterface $outerRepository = null)
```

Comment:
Updated in https://github.com/ezsystems/ezpublish-kernel/pull/1720/commits/bffc066c1c0e679a97347739b506965fbafc789f (`Repository::sudo()` was actually not deprecated)